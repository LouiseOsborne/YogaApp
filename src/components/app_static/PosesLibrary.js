import React, { useState, useEffect } from "react";
import axios from "axios";
import "../../styles/app_static_sass/poses-library.scss";
import TopNavBar from "../nav_and_search/TopNavBar";
import SideNavBar from "../nav_and_search/SideNavBar";
import PosePreview from "./PosePreview";

const PosesLibrary = () => {
  const [poses, setPoses] = useState([]);
  const [selectedCategory, setSelectedCategory] = useState(null);
  const handleSelectedCategory = (category) => {
    setSelectedCategory(category);
  };
  const handleClearFilter = () => {
    setSelectedCategory(null);
  };

  useEffect(() => {
    let endpoint = "/poses";

    if (selectedCategory) {
      endpoint = `/poses/category/${selectedCategory}`;
    }

    axios.get(endpoint).then((response) => {
      const yogaPosesArray = response.data;
      const yogaPictures = yogaPosesArray.map((pose) => [
        pose.url_png,
        pose.pose_name,
        pose.sanskrit_name,
        pose.pose_benefits,
        pose.pose_description,
      ]);
      setPoses(yogaPictures);
    });
  }, [selectedCategory]);

  return (
    <div>
      <div className="top-nav-bar">
        <TopNavBar />
      </div>
      <div className="side-bar">
        <SideNavBar
          handleSelectedCategory={handleSelectedCategory}
          handleClearFilter={handleClearFilter}
        />
      </div>
      <div className="page-content__container">
        <div className="poses__library">
          <div className="poses__container">
            {poses.map((pose, index) => (
              <div className="grid-item" key={index} onClick={() => {}}>
                <PosePreview
                  poseImage={pose[0]}
                  englishName={pose[1]}
                  sanskritName={pose[2]}
                  poseBenefits={pose[3]}
                  poseDescription={pose[4]}
                />
              </div>
            ))}
          </div>
        </div>
      </div>
    </div>
  );
};

export default PosesLibrary;
