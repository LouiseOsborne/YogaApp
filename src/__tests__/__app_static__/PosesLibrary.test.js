import React from "react";
import { render, screen, waitFor, fireEvent } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";
import axios from "axios";
import PosesLibrary from "../../components/app_static/PosesLibrary";

describe("PosesLibrary", () => {
  it("renders correctly", () => {
    const { asFragment } = render(
      <MemoryRouter>
        <PosesLibrary />
      </MemoryRouter>
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
