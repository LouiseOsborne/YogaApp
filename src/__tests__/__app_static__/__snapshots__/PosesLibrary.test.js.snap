// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PosesLibrary renders correctly 1`] = `
<DocumentFragment>
  <div>
    <div
      class="top-nav-bar"
    />
    <div
      class="side-bar"
    >
      <div
        class="sidebar-container"
      >
        <div
          class="search-bar"
        >
          <form>
            <input
              class="search-input"
              placeholder="Search by name ..."
              type="text"
            />
            <button
              class="clear-filter onClick={handleClearFilter}"
              type="submit"
            >
              Clear filter
            </button>
            <button
              class="search-button"
              type="submit"
            >
              Search
            </button>
          </form>
        </div>
        <div
          class="sidebar-title"
        >
          Categories
        </div>
        <div
          class="sidebar-div"
        >
          <button
            class="sidebar-button"
          >
            Core
          </button>
        </div>
        <div
          class="sidebar-div"
        >
          <button
            class="sidebar-button"
          >
            Seated
          </button>
        </div>
        <div
          class="sidebar-div"
        >
          <button
            class="sidebar-button"
          >
            Strengthening
          </button>
        </div>
        <div
          class="sidebar-div"
        >
          <button
            class="sidebar-button"
          >
            Chest Opening
          </button>
        </div>
        <div
          class="sidebar-div"
        >
          <button
            class="sidebar-button"
          >
            Backbend
          </button>
        </div>
        <div
          class="sidebar-title"
        >
          Skill level
        </div>
        <div
          class="sidebar-div"
        >
          <button
            class="sidebar-button"
          >
            Beginner
          </button>
        </div>
        <div
          class="sidebar-div"
        >
          <button
            class="sidebar-button"
          >
            Intermediate
          </button>
        </div>
        <div
          class="sidebar-div"
        >
          <button
            class="sidebar-button"
          >
            Advanced
          </button>
        </div>
      </div>
    </div>
    <div
      class="page-content__container"
    >
      <div
        class="poses__library"
      >
        <div
          class="poses__container"
        />
      </div>
    </div>
  </div>
</DocumentFragment>
`;
